activation/handler_v2.go

```diff
@@ -603,6 +614,39 @@ func (h *HandlerV2) checkMalicious(
 	return false, nil, nil
 }
 
+func (h *HandlerV2) checkDoubleMarry(
+	tx *sql.Tx,
+	watx *wire.ActivationTxV2,
+) (*mwire.MalfeasanceProof, error) {
+	checkMarried := func(tx *sql.Tx, id types.NodeID) (*mwire.MalfeasanceProof, error) {
+		married, err := identities.Married(tx, id)
+		if err != nil {
+			return nil, fmt.Errorf("checking if ID is married: %w", err)
+		}
+		if married {
+			proof := &mwire.MalfeasanceProof{
+				Proof: mwire.Proof{
+					Type: mwire.DoubleMarry,
+					Data: &mwire.DoubleMarryProof{},
```

Comment:
Note: the proof must be implemented first to fill it here.

Reply:
This would also create one proof for every identity with a certificate in the ATX which is not necessary. There should only be one proof that one identity married twice - and a list of certificates that all married that identity and become malicious too.